server/text_generation_server/models/causal_lm.py

```diff
@@ -630,6 +642,52 @@ def __init__(
             self.hb_profer_started = False
         self.step = 0
 
+    def setup_env(self):
+        import sys
+        assert "habana_frameworks" not in sys.modules
```

Comment:
you are importing the package multiple times in the script, it will break by defnition so no need to do additional assert in here

Reply:
please take closer look, the assertion is to make sure "habana_frameworks" is not loaded before those env variables are set